### PR TITLE
Cleanup cyclomatic complexity with function EnsureWork

### LINKED ISSUE #460: Cleanup cyclomatic complexity issue
**What would you like to be added**:
There are 4 functions in which cyclomatic complexity exceed 15:
- [x] 1. EnsureWork() with complexity 21: (@XiShanYongYe-Chang #473)
https://github.com/karmada-io/karmada/blob/dba61fa92e8d22fb2292f5e5cd50d188185d7a34/pkg/util/helper/binding.go#L154
- [x] 2. retainServiceFields() with complexity 20: (@XiShanYongYe-Chang #476 )
https://github.com/karmada-io/karmada/blob/dba61fa92e8d22fb2292f5e5cd50d188185d7a34/pkg/util/helper/binding.go#L154
- [x] 3. getScheduleType() with complexity 20: (@ESonata #685)
https://github.com/karmada-io/karmada/blob/dba61fa92e8d22fb2292f5e5cd50d188185d7a34/pkg/scheduler/scheduler.go#L323
- [x] 4. RunJoin() with complexity (@XiShanYongYe-Chang #636 )
https://github.com/karmada-io/karmada/blob/dba61fa92e8d22fb2292f5e5cd50d188185d7a34/pkg/karmadactl/join.go#L154

You can get the complexity on your side by following commands:
```
# go get github.com/fzipp/gocyclo/cmd/gocyclo@latest
# gocyclo -over 15 pkg
21 helper EnsureWork pkg/util/helper/binding.go:154:1
20 objectwatcher retainServiceFields pkg/util/objectwatcher/retain.go:61:1
20 scheduler (*Scheduler).getScheduleType pkg/scheduler/scheduler.go:323:1
18 karmadactl RunJoin pkg/karmadactl/join.go:154:1
```

I'd prefer to fix it by separated PRs(each PR fix one issue), forks can choose any issues.

**Why is this needed**:
A function with a higher cyclomatic complexity requires more test cases to cover all possible paths and is potentially harder to understand.

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Clear functions with cyclomatic complexity greater than 15.

EnsureWork() with complexity 21:

https://github.com/karmada-io/karmada/blob/dba61fa92e8d22fb2292f5e5cd50d188185d7a34/pkg/util/helper/binding.go#L154

**Which issue(s) this PR fixes**:
Part of #460

### CODE REVIEW

**Comment by RainbowMango** [comment]
Please don't do all these things in one PR, it's hard to review.

**Comment by RainbowMango** [comment]
re-triggered the CI workflow.

**Comment by RainbowMango** [comment]
/hold
Please squash commits.

File: pkg/util/helper/binding.go [hunk 1]
```diff
@@ -162,54 +160,38 @@ func FetchWorkload(dynamicClient dynamic.Interface, restMapper meta.RESTMapper,
...
+		targetClusters = bindingObj.Spec.Clusters
 	}
 
-	var workLabel = make(map[string]string)
+	hasScheduledReplica, referenceRSP, desireReplicaInfos, err := getRSPAndReplicaInfos(c, workload, targetClusters)
+	if err != nil {
+		return err
+	}
 
+	clusterNames := GetBindingClusterNames(targetClusters)
```
Comment (RainbowMango) [inline pkg/util/helper/binding.go:179]: You can just range the `targetClusters` like this:
```golang
for _, targetCluster := range targetClusters{
```

Comment (XiShanYongYe-Chang) [reply pkg/util/helper/binding.go]: I have modified it.

**Pull request merged**